src/vcf/record.cpp

```diff
@@ -340,30 +341,78 @@ namespace ebi
         }
     }
 
-    void Record::strict_validation_info_predefined_tags(std::string const & field_key, std::string const & field_value) const
+    void Record::strict_validation_info_predefined_tags(std::string const & field_key, std::string const & field_value,
+                                                        std::vector<std::string> const & values) const
     {
         if (field_key == "AA") {
             static boost::regex aa_regex("((?![,;=])[[:print:]])+");
             if (!boost::regex_match(field_value, aa_regex)) {
                 throw new InfoBodyError{line, "INFO AA=" + field_value + " value is not a single dot or a string of bases", field_key};
             }
         } else if (field_key == "AF") {
-            std::vector<std::string> values;
-            util::string_split(field_value, ",", values);
             for (auto & value : values) {
                 if (std::stold(value) < 0 || std::stold(value) > 1) {
                     throw new InfoBodyError{line, "INFO AF=" + field_value + " value does not lie in the interval [0,1]", field_key};
                 }
             }
         } else if (field_key == "CIGAR") {
-            std::vector<std::string> values;
-            util::string_split(field_value, ",", values);
             static boost::regex cigar_string("([0-9]+[MIDNSHPX])+");
             for (auto & value : values) {
                 if (!boost::regex_match(value, cigar_string)) {
                     throw new InfoBodyError{line, "INFO CIGAR=" + field_value + " value is not an alphanumeric string compliant with the SAM specification", field_key};
                 }
             }
+        } else if (field_key == "END") {
+            auto it = info.find("IMPRECISE");
+            if (it != info.end() && it->second == "0") {
+                auto expected = std::to_string(position + reference_allele.length() - 1);
+                if (field_value != expected) {
+                    throw new InfoBodyError{line, "INFO END=" + field_value + " value must be equal to \"POS + length of REF - 1\" for a precise variant (where IMPRECISE is set to 0), expected " + expected, field_key};
+                }
+            }
+        } else if (field_key == "SVLEN" && values.size() == alternate_alleles.size()) {
+            bool is_ref_symbolic = true;
+            std::vector<bool> is_alt_symbolic(alternate_alleles.size(), false);
+            std::unordered_set<char> symbolic = { 'A', 'C', 'G', 'T', 'N', 'a', 'c', 'g', 't', 'n' };
+
+            for (auto & ref : reference_allele) {
+                if (symbolic.find(ref) == symbolic.end()) {
+                    is_ref_symbolic = false;
+                    break;
+                }
+            }
+
+            if (is_ref_symbolic) {
+                for (auto i = 0; i < alternate_alleles.size(); i++) {
```

Comment:
small warning here and in line 397: `auto i` is deduced to int (signed), and `alternate_alleles.size()` is size_t (unsigned). I usually prefer using `size_t i` directly. As a rule of thumb, if a variable is for counting (and non-negative) size_t is a good type.